elixir/markdown/lib/markdown.ex

```diff
@@ -11,37 +11,53 @@ defmodule Markdown do
     "<h1>Header!</h1><ul><li><em>Bold Item</em></li><li><i>Italic Item</i></li></ul>"
   """
 
-  #This should probably be a pipeline instead of all those nested functions
+  #Refactored into a pipeline to get rid of the nested functions
   @spec parse(String.t()) :: String.t()
   def parse(m) do
-    patch(Enum.join(Enum.map(String.split(m, "\n"), fn t -> process(t) end)))
-  end
+    m
+    |> String.split("\n")
+    |> IO.inspect(label: "output of String.split")
+    |> Enum.map(& process(&1))
+    |> IO.inspect(label: "Enum.map process output")
+    |> Enum.join()
+    |> IO.inspect(label: "output of Enum.join")
+    |> patch()
 
+  end
 
-  defp process(t) do
-    case String.first(t) do
-      "#" -> enclose_with_header_tag(parse_header_md_level(t))
-      "*" -> parse_list_md_level(t)
-      _ -> enclose_with_paragraph_tag(String.split(t))
+  #I removed the nested if structure and replaced it with 'case'"
+  # changed 't' into 'line' because it is the result of String.split in the parse function
+  defp process(line) do
+    case String.first(line) do
+      "#" -> enclose_with_header_tag(parse_header_md_level(line))
+      "*" -> parse_list_md_level(line)
+      _ -> enclose_with_paragraph_tag(String.split(line))
     end
   end
 
-  defp parse_header_md_level(hwt) do
-    [h | t] = String.split(hwt)
+  #decided to change the 'hwt' argument to 'line' since it is still the line from the process function that gets passed in here
+  defp parse_header_md_level(line) do
+    [h | t] = String.split(line)
     {to_string(String.length(h)), Enum.join(t, " ")}
+    |> IO.inspect(label: "output of parse_header_md_level")
+    # {"number", "text minus the header markup"}
```

Comment:
Is this a forgotten comment?

One thing that might be able to improve the readability of line 41 is some well-named variables.

Reply:
?

Reply:
The forgotten comment is deleted and I have changed the variables.

Reply:
:+1: